src/main/java/com/uwetrottmann/tmdb2/TmdbInterceptor.java

```diff
@@ -39,10 +70,95 @@ public static Response handleIntercept(Chain chain, String apiKey) throws IOExce
         HttpUrl.Builder urlBuilder = request.url().newBuilder();
         urlBuilder.setEncodedQueryParameter(Tmdb.PARAM_API_KEY, apiKey);
 
+        if (request.url().pathSegments().get(0).equals("account")) {
```

Comment:
Thanks, you are right! I should have had a closer look at the interceptor used in that project.

The `authenticateAnonymously()` and `authenticate()` code added to `Tmdb` should be moved to an authenticator instead. And the interceptor should then not throw exceptions if there is no auth data, the authenticator should if it fails.